src/Microsoft.DotNet.ProjectModel/ProjectSystem/ProjectSystem.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.ProjectModel;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.ProjectModel.ProjectSystem
+{
+    public class ProjectSystem
+    {
+        private readonly HashSet<string> _projectPaths = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+        private readonly ProjectSystemCache _cache = new ProjectSystemCache();
+
+        public ProjectSystem(string projectPath)
```

Comment:
I'm not a fan of constructors that do I/O. Generally constructors should be "failure-proof" pure operations (as much as possible). Can we refactor this into a constructor that takes the list of project paths (possibly private) and a public static factory method? Something like `ProjectSystem.Create(projectPath)`